ExternalLookup/src/main/java/cheetah/example/externallookup/function/ExternalLookupMapper.java

```diff
@@ -23,13 +25,16 @@ public class ExternalLookupMapper extends RichAsyncFunction<InputEvent, OutputEv
 
     private String idServiceHost;
     private HttpClient client;
+    private KeyedTokenProvider tokenProvider;
+    private final static String TOKEN_ID = "ServiceToken";
 
     @Override
     public void asyncInvoke(InputEvent externalLookupInputEvent, ResultFuture<OutputEvent> resultFuture) {
 
         HttpRequest request = HttpRequest.newBuilder()
```

Comment:
the builder can be instantiated in the open function with GET(), uri, and content-type + accept headers and asyncInvoke can then call baseRequest.copy().header("Authorization", ...).build()